stackslib/src/chainstate/stacks/transaction.rs

```diff
@@ -182,6 +234,10 @@ impl StacksMessageCodec for TransactionPayload {
                     }
                 }
             }
+            TransactionPayload::TenureChange(tc) => {
+                write_next(fd, &(TransactionPayloadID::TenureChange as u8))?;
+                tc.consensus_serialize(fd)?;
+            }
```

Comment:
You're going to somehow need to epoch-gate `consensus_deserialize` below so that this new transaction variant does not even decode until epoch 3.0.  I think we're going to need a separate trait for this (see here: https://github.com/stacks-network/stacks-blockchain/pull/3710#issuecomment-1747137322)

Reply:
Yeah this makes sense. I'll check with Vlad to see if he's working on this for his PR

Reply:
Right now, there are at least two PRs in flight (Vlad's and Cyle's) that do this.  We should just bite the bullet and get one of them shipped, and then base this PR off of it.

Reply:
Actually, @kantai has already done this in https://github.com/stacks-network/stacks-blockchain/pull/3992.  Can you rebase off of this?

Reply:
Hey @jbencin @jcnelson ! Can't find in #3992 the logic for `DeserializeWithEpoch` traits for StacksTransaction and StacksBlock, I'm currently working on it and will finish soon, do you think I should wait for that PR to implement this logic and drop mine?

Reply:
No -- #3992 isn't going to implement that logic -- so you should continue with your work.

Reply:
Got it, thanks @kantai

Reply:
Just to close the loop on this PR: I don't think it's strictly necessary to epoch gate *this* deserialization. All that matters is whether or not the transaction is valid for inclusion in a block. During block processing, the transaction processor should just yield an invalid block error if this transaction payload is matched.

Reply:
Yeah that's fine for now.  Furthermore, the block validation code reached from `accept_block()` can reject blocks that include them too early.  I'm already doing this in my coordinator branch.